File: 프로그래머스/1/131705. 삼총사/삼총사.js
function solution(number) {
    var ans = 0;
    const getCombinations = function (arr, selectNumber) {
        const results = [];
        if (selectNumber === 1) return arr.map((el) => [el]); 
        // n개중에서 1개 선택할 때(nC1), 바로 모든 배열의 원소 return

        arr.forEach((fixed, index, origin) => {
            const rest = origin.slice(index + 1); 
            const combinations = getCombinations(rest, selectNumber - 1); 
            const attached = combinations.map((el) => [fixed, ...el]); 
            results.push(...attached); 
        });

        return results;
    }
    
    var comb = getCombinations(number, 3);
    for (c of comb) {
        var sum = 0;
        c.forEach((num) => {
            sum += num;
        })
        if (sum == 0)
            ans += 1;
    }
    return ans;
}

File: 프로그래머스/1/131705. 삼총사/삼총사.py
from itertools import combinations

def solution(number):
    combination = combinations(number, 3)  # 3명의 정수 번호 조합
    
    # 3명의 정수 번호 합이 0일 경우(= 삼총사일 경우) +1
    return sum(list(1 if sum(combi) == 0 else 0 for combi in combination))